src/components/TokenSelect/TokenButton.tsx

```diff
@@ -13,13 +13,17 @@ const transitionCss = css`
   transition: background-color 0.125s linear, border-color 0.125s linear, filter 0.125s linear, width 0.125s ease-out;
 `
 
-const StyledTokenButton = styled(Button)`
+const StyledTokenButton = styled(Button)<{ approved?: boolean }>`
   border-radius: ${({ theme }) => theme.borderRadius}em;
   padding: 0.25em;
 
   :enabled {
     ${({ transition }) => transition && transitionCss};
   }
+
+  ${TokenImg} {
+    filter: ${({ approved }) => approved === false && 'grayscale(1)'};
+  }
```

Comment:
I prefer to have some kind of explicit unset value here instead of creating a floating invalid `filter` rule: `filter: `